### ISSUE TITLE
GitHub Integration: Unable to install GitHub Integration

### ISSUE THREAD

**Issue by roman-stelmakh-emesa** [getsentry/sentry#18335]
I tried to install GitHub Integration on on-premise Sentry using this page https://docs.sentry.io/server/integrations/github/
After config.yml was changed and Sentry was reloaded, I went to **/settings/${organization}/integrations** and clicked Install GitHub Integration. I was redirected  to my GitHub app, and I installed it and allowed all repositories.
Then I was redirected back to the Sentry home page. 
There is no GitHub Integration on the integrations page, and a button Install is still there.
However GitHub App show successful payload deliveries to **/extensions/github/webhook/** and their statuses are always 204.

**Comment by jakana-imb** [comment]
@roman-stelmakh-emesa have you had any luck with this?

**Comment by roman-stelmakh-emesa** [comment]
@jakana-imb No, the problem still exists . I also set webhook secret, but it didn't help.

**Comment by jakana-imb** [comment]
Thanks for getting back to me, same here. I'll post back if we ever get it working.

**Comment by dehy** [comment]
I have exactly the same problem. the /extensions/github/setup/ page redirect then to the homepage without any further setup.

**Comment by dehy** [comment]
I fixed my problem.

Unlike the official documentation says (which may be for sentry 10.0), I changed my `User authorization callback URL` to `${urlPrefix}/extensions/github/setup/`, as indicated in this issue comment https://github.com/getsentry/sentry/issues/12670#issuecomment-484165388

**Comment by jakana-imb** [comment]
Interesting. I just tried that and get an error when the app redirects to the auth callback.

**Comment by robbi5** [comment]
Had the same problem while trying to configure SSO with GitHub, logs showed `{"message":"Resource not accessible by integration","documentation_url":"https://developer.github.com/v3/users/emails/#list-email-addresses-for-a-user"}`.

Turns out, the required `user:email` scope seems not to be usable with an GitHub App, as [the source](https://github.com/getsentry/sentry/blob/988a35af72efff0e618d882f2c996dc5a554313a/src/sentry/identity/github/provider.py#L23-L26) says.

I finally had to create a seperate GitHub OAuth Integration and put the credentials from there as `GITHUB_APP_ID` and `GITHUB_API_SECRET` into `sentry/sentry.conf.py`.

**Comment by djgalloway** [comment]
> Had the same problem while trying to configure SSO with GitHub, logs showed `{"message":"Resource not accessible by integration","documentation_url":"https://developer.github.com/v3/users/emails/#list-email-addresses-for-a-user"}`.
> 
> Turns out, the required `user:email` scope seems not to be usable with an GitHub App, as [the source](https://github.com/getsentry/sentry/blob/988a35af72efff0e618d882f2c996dc5a554313a/src/sentry/identity/github/provider.py#L23-L26) says.
> 
> I finally had to create a seperate GitHub OAuth Integration and put the credentials from there as `GITHUB_APP_ID` and `GITHUB_API_SECRET` into `sentry/sentry.conf.py`.

Did you create another "GitHub App" or just an OAuth App?  Asking because I was able to get GitHub integration working by using a GitHub App but can't get SSO going.

**Comment by BYK** [comment]
@djgalloway the instructions on https://develop.sentry.dev/self-hosted/sso/#github-auth should set this up for both. Do you think they are accurate or misleading?

**Comment by djgalloway** [comment]
> @djgalloway the instructions on https://develop.sentry.dev/self-hosted/sso/#github-auth should set this up for both. Do you think they are accurate or misleading?

I think what happened was I upgraded my on-premise sentry over a month ago and when I tried finding the GitHub SSO docs, I didn't find any.  I hadn't bothered to search again the second time around I tried setting it up (last week) and by then, the docs had been written and [committed](https://github.com/getsentry/develop/commit/220f257c606ee8405dc6767e649f83b6f2610402) :smile: 

The docs are perfect though.  Thank you!
